tactics/elim.ml

```diff
@@ -202,14 +190,10 @@ let general_decompose recognizer c =
   end
 
 let head_in indl t gl =
-  let env = Proofview.Goal.env gl in
   let sigma = Tacmach.New.project gl in
   try
-    let ity,_ =
-      if !up_to_delta
-      then find_mrectype env sigma t
-      else extract_mrectype sigma t
```

Comment:
So, you renounce to explore the "better" up-to-normalization strategy? Maybe compensate this by adding a comment on the code to tell that we would expect a more natural user experience by reasoning up-to-normalization but that this is questionable for compatibility?

Reply:
This is dead code that is not reachable by any means, and if ever we wanted to tweak the behaviour it'd be trivial codewise (although probably not compatibility-wise).  Thus I don't really see the interest to keep it around.

Reply:
Actually, this is only used by the `decompose` tactic. I don't really understand why we still have such a tactic in 2020 when it's easy to replace with an Ltac snippet. I'd be curious how much these tactics are used.

Reply:
It seems to be used in VST, hoare-tut, exact-real-arithmetic, GeoCoq, CompCert, color, ... Also in the stdlib.